Source/WTF/wtf/generic/WorkQueueGeneric.cpp

```diff
@@ -41,16 +41,14 @@ WorkQueueBase::WorkQueueBase(RunLoop& runLoop)
 
 void WorkQueueBase::platformInitialize(const char* name, Type, QOS qos)
 {
-    BinarySemaphore semaphore;
-    Thread::create(name, [&] {
+    m_runLoop = RunLoop::create(name, ThreadType::Unknown, qos).ptr();
 #if ASSERT_ENABLED
+    BinarySemaphore semaphore;
+    m_runLoop->dispatch([&] {
         m_threadID = Thread::current().uid();
-#endif
-        m_runLoop = &RunLoop::current();
-        semaphore.signal();
-        m_runLoop->run();
-    }, ThreadType::Unknown, qos)->detach();
+    });
```

Comment:
Semaphore's not signaled.

Reply:
Thanks!
(Appears that work queue tests aren't run WorkQueueGeneric platforms?)

Reply:
Only layout tests are run on the GTK tester EWS, for some reason.